### PR TITLE
Mavlink wait for boot complete before sending or receiving anything

### PR DESCRIPTION
**Pull request by dagar**
This doesn't necessarily fix anything, but I feel a bit more comfortable not allowing any mavlink work until boot is complete.

### CODE REVIEW

File: src/modules/mavlink/mavlink_main.cpp [hunk 1]
```diff
@@ -2208,6 +2202,11 @@ Mavlink::task_main(int argc, char *argv[])
 		/* main loop */
 		px4_usleep(_main_loop_delay);
 
+		// If the wait until transmit flag is on, only transmit after we've received messages
```
Comment (julianoes) [inline src/modules/mavlink/mavlink_main.cpp]: I don't  think this comment is very helpful. I'd just remove it and if someone wants to understand they have to look into the `should_transmit()` method.

Comment (dagar) [reply src/modules/mavlink/mavlink_main.cpp]: Honestly I think my brain strips comments before processing.

**Comment by dannyfpv** [comment]
tested on pixhawk4 v5 f-450

on mission mode
altitude mode
no issues

log:
https://review.px4.io/plot_app?log=739d1934-829f-46f8-a447-9291e084c082

**Comment by jorge789** [comment]
**Tested on PixRacer V4:**
**Modes Tested**

Position Mode: Good.
Altitude Mode: Good.
Stabilized Mode: Good.
Mission Plan Mode (Automated): Good.
RTL: Good.

**- Procedure**
Arm and Take off in position mode, after flying for approximately one minute, switched to altitude then stabilized mode proceed to switch to mission plan mode then make sure that vehicle follows all waypoints as shown in QGC, once completed all waypoin activate RTL and see landing behaviour.

**Notes:**
No issues noted, good flight in general.

**Log:**
https://review.px4.io/plot_app?log=d0e87d88-f61f-4dc1-aefc-28432b42e020

**Tested on CUAV+ V5:**
**Modes Tested**

Position Mode: Good.
Altitude Mode: Good.
Stabilized Mode: Good.
Mission Plan Mode (Automated): Good.
RTL: Good.

**- Procedure**
Arm and Take off in position mode, after flying for approximately one minute, switched to altitude then stabilized mode proceed to switch to mission plan mode then make sure that vehicle follows all waypoints as shown in QGC, once completed all waypoin activate RTL and see landing behaviour.

**Notes:**
No issues noted, good flight in general.

**Log:**
https://review.px4.io/plot_app?log=e8ad9e2f-4d20-46e2-bc46-2b1adfa0892e

**Pull request merged**